src/include/homestore/replication_service.hpp

```diff
@@ -6,18 +6,31 @@
 
 #include <folly/futures/Future.h>
 
-#include "repl_decls.h"
-#include "repl_set.h"
-
-namespace nuraft {
-class state_machine;
-}
+#include <homestore/replication/repl_decls.h>
 
 namespace homestore {
 
+// clang-format off
+VENUM(ReplServiceError, int32_t,
+      OK = 0,         // Everything OK
+      CANCELLED = -1, // Request was cancelled
+      TIMEOUT = -2, 
+      NOT_LEADER = -3, 
+      BAD_REQUEST = -4, 
+      SERVER_ALREADY_EXISTS = -5, 
+      CONFIG_CHANGING = -6,
+      SERVER_IS_JOINING = -7, 
+      SERVER_NOT_FOUND = -8, 
+      CANNOT_REMOVE_LEADER = -9, 
+      SERVER_IS_LEAVING = -10,
+      TERM_MISMATCH = -11, 
+      RESULT_NOT_EXIST_YET = -10000, 
+      NOT_IMPLEMENTED = -10001,
+      FAILED = -32768);
```

Comment:
`INT32_MIN`? this I believe is `== INT32_MAX` as is.

Reply:
I just copied from nuraft repl_service error. Do you want me to change that I can do that.

Reply:
Nah; should prolly really be `std::numeric_limits<ENUM>::min();`; but no need.